rust/src/datatypes.rs

```diff
@@ -72,14 +75,16 @@ impl Schema {
             .enumerate()
             .find(|&(_,c)| c.name == name)
     }
+}
 
-    pub fn to_string(&self) -> String {
-        let s : Vec<String> = self.columns.iter()
+impl fmt::Display for Schema {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let s: Vec<String> = self.columns
```

Comment:
Very minor, but you could consider using `write!`, like you added, e.g. 
```rust
fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
  for (i, col) in self.columns.iter().enumerate() {
    write!(f, "{}", col)?;
    if i < self.columns.len() - 1 {
      write!(f, ", ")?;
    }
  }
  Ok(())
}
```

Reply:
OK great, thanks for the pointer, I'll make that change.

Why do we need the `?` at the end of the lines there?